src/lines-component.coffee

```diff
@@ -213,8 +213,7 @@ LinesComponent = React.createClass
     innerHTML = ""
 
     scopeStack = []
-    firstTrailingWhitespacePosition = text.search(/\s*$/)
-    lineIsWhitespaceOnly = firstTrailingWhitespacePosition is 0
+    lineIsWhitespaceOnly = line.isOnlyWhitespace()
```

Comment:
FAO reviewer: This is the main issue. After substituteInvisibleCharacters has run the line text is no longer whitespace only and contains the invisible character glyphs.

My fix removes the regex check and simply calls isOnlyWhitespace on the TokenizedLine, which remembers if the original text in the buffer is whitespace only or not.